libs/dal/src/repositories/tenant/tenant.schema.ts

```diff
@@ -23,6 +23,21 @@ tenantSchema.index({
   _environmentId: 1,
 });
 
+/*
+ * This index was initially created to optimize:
+ *
+ * Path: apps/api/src/app/tenant/usecases/create-tenant/create-tenant.usecase.ts
+ * Context: execute()
+ * Query: findOne({
+ *    _environmentId: command.environmentId,
+ *    identifier: command.identifier,
+ *  });
+ */
+tenantSchema.index({
+  identifier: 1,
+  _environmentId: 1,
+});
```

Comment:
Created to the new use case.

Reply:
I would suggest to make this compound index unique if this combination is supposed to be unique (if I understand correctly).

Even with `if` statement in create usecase, some weird race conditions are possible.

Reply:
Thanks, love the suggestion, I think it would be a very edge case but still, there is a possibility it could happen. I will take a look at it.